src/contracts/libraries/SlashingLib.sol

```diff
@@ -135,6 +138,9 @@ library SlashingLib {
             .divWad(slashingFactor);
 
         dsf._scalingFactor = newDepositScalingFactor;
+        
+        // Avoid potential underflow.
+        require(newDepositScalingFactor != 0, InvalidDepositScalingFactor());
```

Comment:
👍 

Just needs a unit test. Here are inputs that hit this case:

Nice. Needs a unit test - I have a specific set of inputs that should hit that case:

```solidity
dsf._scalingFactor = 1
dsf.update({
    prevDepositShares: 1e18,
    addedShares: 1,
    slashingFactor: WAD - 1
});
```